modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-impl/src/main/java/com/liferay/headless/commerce/admin/order/internal/resource/v1_0/OrderItemResourceImpl.java

```diff
@@ -288,6 +288,155 @@ public OrderItem postOrderIdOrderItem(Long id, OrderItem orderItem)
+			commerceOrderItem.getCommerceOrderItemId(), orderItem.getOptions(),
+			orderItem.getQuantity(),
+			_commerceContextFactory.create(
+				contextCompany.getCompanyId(), commerceOrder.getGroupId(),
+				contextUser.getUserId(), commerceOrder.getCommerceOrderId(),
+				commerceOrder.getCommerceAccountId()),
+			_serviceContextHelper.getServiceContext(
+				commerceOrderItem.getGroupId()));
+
+		// Pricing
+
+		PortletResourcePermission portletResourcePermission =
+			_commerceOrderModelResourcePermission.
+				getPortletResourcePermission();
+
+		if (portletResourcePermission.contains(
+				PermissionThreadLocal.getPermissionChecker(),
+				commerceOrder.getGroupId(),
+				CommerceActionKeys.MANAGE_COMMERCE_ORDER_PRICES)) {
+
+			commerceOrderItem =
+				_commerceOrderItemService.updateCommerceOrderItemPrices(
+					commerceOrderItem.getCommerceOrderItemId(),
+					orderItem.getDiscountAmount(),
+					orderItem.getDiscountWithTaxAmount(),
+					orderItem.getDiscountPercentageLevel1(),
+					orderItem.getDiscountPercentageLevel1WithTaxAmount(),
+					orderItem.getDiscountPercentageLevel2(),
+					orderItem.getDiscountPercentageLevel2WithTaxAmount(),
+					orderItem.getDiscountPercentageLevel3(),
+					orderItem.getDiscountPercentageLevel3WithTaxAmount(),
+					orderItem.getDiscountPercentageLevel4(),
+					orderItem.getDiscountPercentageLevel4WithTaxAmount(),
+					orderItem.getFinalPrice(),
+					orderItem.getFinalPriceWithTaxAmount(),
+					orderItem.getPromoPrice(),
+					orderItem.getPromoPriceWithTaxAmount(),
+					orderItem.getUnitPrice(),
+					orderItem.getUnitPriceWithTaxAmount());
+		}
+
+		// Expando
+
+		Map<String, ?> customFields = _getExpandoBridgeAttributes(orderItem);
+
+		if ((customFields != null) && !customFields.isEmpty()) {
+			ExpandoUtil.updateExpando(
+				contextCompany.getCompanyId(), CommerceOrderItem.class,
+				commerceOrderItem.getPrimaryKey(), customFields);
+		}
+
+		return _toOrderItem(commerceOrderItem.getCommerceOrderItemId());
+	}
+
+	@Override
+	public OrderItem putOrderItemByExternalReferenceCode(
+			String externalReferenceCode, OrderItem orderItem)
+		throws Exception {
+
+		CommerceOrder commerceOrder = _commerceOrderService.getCommerceOrder(
+			orderItem.getOrderId());
+
+		CommerceOrderItem commerceOrderItem =
+			_commerceOrderItemService.fetchByExternalReferenceCode(
+				externalReferenceCode, contextCompany.getCompanyId());
+
+		if (commerceOrderItem == null) {
+			commerceOrderItem = OrderItemUtil.addCommerceOrderItem(
+				_cpInstanceService, _commerceOrderItemService,
+				_commerceOrderModelResourcePermission, orderItem, commerceOrder,
+				_commerceContextFactory.create(
+					contextCompany.getCompanyId(), commerceOrder.getGroupId(),
+					contextUser.getUserId(), commerceOrder.getCommerceOrderId(),
+					commerceOrder.getCommerceAccountId()),
+				_serviceContextHelper.getServiceContext(
+					commerceOrder.getGroupId()));
+
+			return _toOrderItem(
```

Comment:
Are we missing to update prices with this one?

Reply:
Are you referring to the OrderItemUtil.addCommerceOrderItem call?

Reply:
Yes, just cheked that the utility class do it directly. Forget about this one

Reply:
That's made in the Util